kuksa_databroker/databroker/src/broker.rs

```diff
@@ -708,6 +880,28 @@ impl Entries {
         // Map name -> id
         self.path_to_id.insert(name.clone(), id);
 
+        if let Some(allowed_values) = &allowed {
+            match (allowed_values, &data_type) {
+                (DataValue::BoolArray(_allowed_values), DataType::Bool) => (),
+                (DataValue::StringArray(_allowed_values), DataType::String) => (),
+                (DataValue::Int32Array(_allowed_values), DataType::Int32) => (),
+                (DataValue::Int64Array(_allowed_values), DataType::Int64) => (),
+                (DataValue::Uint32Array(_allowed_values), DataType::Uint32) => (),
+                (DataValue::Uint64Array(_allowed_values), DataType::Uint64) => (),
+                (DataValue::FloatArray(_allowed_values), DataType::Float) => (),
+                (DataValue::DoubleArray(_allowed_values), DataType::Double) => (),
+                (DataValue::BoolArray(_allowed_values), DataType::BoolArray) => (),
+                (DataValue::StringArray(_allowed_values), DataType::StringArray) => (),
+                (DataValue::Int32Array(_allowed_values), DataType::Int32Array) => (),
+                (DataValue::Int64Array(_allowed_values), DataType::Int64Array) => (),
+                (DataValue::Uint32Array(_allowed_values), DataType::Uint32Array) => (),
+                (DataValue::Uint64Array(_allowed_values), DataType::Uint64Array) => (),
+                (DataValue::FloatArray(_allowed_values), DataType::FloatArray) => (),
+                (DataValue::DoubleArray(_allowed_values), DataType::DoubleArray) => (),
+                _ => return Err(RegistrationError {}),
+            }
+        }
+
```

Comment:
This looks very much like the `fn validate_allowed_type(..)` above.
You can probably reuse it by mapping the errors, i.e.
```
validate_allowed_type(..).map_err(|err| RegistrationsError {})?;
```

Maybe a `ValidationError` should be introduced since the errors are not only for updates anymore.

Reply:
This then should be moved out of the implementation of Entry. Into DataValue makes not much sense in my opinion, because it is a corner case for a DataValue. Any suggestions where it could fit or just standalone?

Reply:
> This then should be moved out of the implementation of Entry.

Why?

What I mean is that it makes sense to reuse validation logic regardless of if we're adding an entry or updating it. It should be validated the same, right?

Reply:
Right. Then the mechanism to update needs to be altered a bit. Okay, I see a way to do it. Thanks for the quick reply!